src/ui/profile_preferences.ui

```diff
@@ -568,27 +568,93 @@
+         </property>
+         <property name="checked">
+          <bool>false</bool>
+         </property>
+         <property name="toolTip">
+          <string>&lt;p&gt;This option enables or disables all spell-checking functionality.&lt;p&gt;</string>
+         </property>
+         <layout class="QVBoxLayout" name="verticalLayout_2">
+          <item>
+           <widget class="QLabel" name="label_chooseSpellCheckDictionary">
             <property name="text">
-             <string>Choose spell check dictionary:</string>
+             <string notr="true" extracomment="This text is not for translation as it is replaced by an OS specific text on dialogue construction.">System/Mudlet dictionary:</string>
             </property>
            </widget>
           </item>
-          <item row="0" column="1">
-           <widget class="QListWidget" name="dictList"/>
-          </item>
-          <item row="1" column="0" colspan="2">
-           <widget class="QCheckBox" name="enableSpellCheck">
-            <property name="text">
-             <string>Enable spell check</string>
+          <item>
+           <widget class="QListWidget" name="dictList">
+            <property name="sizeAdjustPolicy">
+             <enum>QAbstractScrollArea::AdjustToContents</enum>
             </property>
-            <property name="checked">
+            <property name="dragDropMode">
+             <enum>QAbstractItemView::DragDrop</enum>
+            </property>
+            <property name="movement">
+             <enum>QListView::Snap</enum>
+            </property>
+            <property name="flow">
+             <enum>QListView::TopToBottom</enum>
+            </property>
+            <property name="resizeMode">
+             <enum>QListView::Adjust</enum>
+            </property>
+            <property name="viewMode">
+             <enum>QListView::IconMode</enum>
+            </property>
+            <property name="sortingEnabled">
              <bool>true</bool>
             </property>
+            <property name="toolTip">
+             <string>&lt;p&gt;Select one dictionary which will be available on the command line and in the lua sub-system.&lt;p&gt;</string>
+            </property>
+           </widget>
+          </item>
+          <item>
+           <widget class="QGroupBox" name="groupBox_userDictionary">
+            <property name="title">
+             <string>User dictionary:</string>
+            </property>
+            <layout class="QHBoxLayout" name="horizontalLayout_userDictionary">
+             <item>
+              <widget class="QRadioButton" name="radioButton_userDictionary_none">
+               <property name="text">
+                <string>None</string>
+               </property>
+               <property name="checked">
+                <bool>true</bool>
+               </property>
+               <property name="toolTip">
+                <string>&lt;p&gt;A user dictionary will not be used.&lt;p&gt;</string>
```

Comment:
Well we didn't have it in the past and maybe some people like it like that... after all *there's nowt so queer as folk*!

Reply:
... and even now I do not know that the user dictionary will work as expected for, say Arabic or RTL scripts. As it happens although spell-checking *is* enabled by default and set for "English (American)" the user dictionary is disabled by default to reflect how the previous versions of Mudlet are - I am sure you will want to draw attention to it in the release announcement... :mega:

Reply:
If it doesn't work people can turn spellchecking off or we'll fix it. I don't think this option is a super-useful one to have - and we already have an "options overload" with so many knobs and fobs, let's stick to adding ones we're super sure are needed?

Reply:
It doesn't seem that sensible to *force* the user to have a personal dictionary (to hold their own curated lists of words for things like persons' names that are never going to fit into a national one) if they don't want it. Don't forget that it will take up a couple of lines on the context menu even if it cannot contribute because it has "no suggestions".  It really doesn't seem that unreasonable to have a three way choice for: shared between all profiles that want it / just for this per profile / not at all to shut it off completely. The latter will make things be how they are before I came up with this plan...

Reply:
It's only how all applications work today - Firefox, LibreOffice and so on. You will not find a 'personal dictionary' option in them!